### PR TITLE
Add breadcrumbs support for Kotest spec files (#5356)

### LINKED ISSUE #5356: Feature request: smarter breadcrumbs in freespec
Would it be possible for the kotest plugin to provide its own breadcrumbs?

When using FreeSpec, the default kotlin breadcrumbs become just a number of anonymous lambdas shown as "{...}".
This is basically useless.

I would like in stead to see the actual case names in the breadcrumbs.

I love FreeSpec, and use it for functional tests suites sort of cucumber-style. 
Those can easily become both long and deep enough that it would be really useful with some help keeping track of exactly which requirement you are looking at.

### PR DESCRIPTION
**Pull request by sksamuel**
## Summary

- Adds `KotestBreadcrumbsCollector`, a `FileBreadcrumbsCollector` that shows the Kotest test hierarchy in the IDE breadcrumb bar: `SpecClass → context scope → test name`.
- Delegates all test-scope detection to `SpecStyle.findAssociatedTest` — no PSI traversal reimplemented in the collector itself.
- `handlesFile` uses a purely syntactic supertype-name check (no Analysis API) so it is safe on background threads.
- `KtClassOrObject.kotestStyleSyntactic()` (in `psi/classes.kt`) maps a class to its `SpecStyle` without type resolution; now recurses through the inheritance hierarchy via `KotlinClassShortNameIndex` so classes extending an intermediate abstract base spec are also recognised.
- Registered in `plugin.xml` as a `fileBreadcrumbsCollector` extension.

## Changes

| File | Change |
|------|--------|
| `KotestBreadcrumbsCollector.kt` | New — collector implementation |
| `psi/classes.kt` | `kotestStyleSyntactic()` — recursive syntactic style detection |
| `plugin.xml` | Register `fileBreadcrumbsCollector` extension |
| `SpecStyle.kt` | Minor: `firstNotNullOfOrNull` idiom (linter) |
| `KotestBreadcrumbsCollectorTest.kt` | New — 30+ tests covering all spec styles and edge cases |
| `KotestStyleSyntacticTest.kt` | New — tests for recursive `kotestStyleSyntactic` |

## Test plan

- [x] `KotestBreadcrumbsCollectorTest` — `handlesFile` (all 10 spec styles + negative cases), `computeCrumbs` for FunSpec, DescribeSpec, BehaviorSpec, StringSpec, FreeSpec, ShouldSpec, FeatureSpec, ExpectSpec, WordSpec, AnnotationSpec, and edge cases
- [x] `KotestStyleSyntacticTest` — direct match, null for non-spec, two-level and three-level transitive inheritance, circular-inheritance guard (no `StackOverflowError`)

Closes #5356

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by sksamuel** [comment]
It looks like this:

image

File: kotest-intellij-plugin/src/main/kotlin/io/kotest/plugin/intellij/styles/SpecStyle.kt [hunk 1]
```diff
@@ -48,12 +48,11 @@ interface SpecStyle {
       fun findTest(element: PsiElement): Test? {
          return styles.asSequence()
             .filter { it.isContainedInSpec(element) }
-            .mapNotNull { it.findAssociatedTest(element) }
-            .firstOrNull()
+            .firstNotNullOfOrNull { it.findAssociatedTest(element) }
```
Comment (alfonsoristorato) [inline kotest-intellij-plugin/src/main/kotlin/io/kotest/plugin/intellij/styles/SpecStyle.kt:51]: of course you'd know the most obscure kotlin utils

**Review by alfonsoristorato — APPROVED** [review]
